### PR TITLE
[WIP]: Jenkinsfile: add Testing stage

### PR DESCRIPTION
**Pull request by arithx**
Adds a testing stage which runs all kola tests for the latest build.
Currently does not archive any testing artifacts.

### CODE REVIEW

**Comment by arithx** [comment]
@dustymabe can you sync up with me on how to test this? I've ran the commands themselves locally inside a `coreos-assembler` container but want to validate it against the actual pipeline.

**Comment by dustymabe** [comment]
> @dustymabe can you sync up with me on how to test this? I've ran the commands themselves locally inside a `coreos-assembler` container but want to validate it against the actual pipeline.

@arithx one way might be to run the environment locally as described in https://github.com/coreos/fedora-coreos-pipeline/blob/a8bc0efdf1c6eba0bd75a03d434c74f473536f52/HACKING.md and see if it works there. We can also set up a container in centos ci and run it and see if it works. we should also look to get you access to that cluster.

File: Jenkinsfile [hunk 1]
```diff
@@ -47,6 +47,14 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
             currentBuild.description = "⚡ ${newBuildID}"
         }
 
+        stage('Test') {
+            utils.shwrap("""
+            latest_build=$(readlink builds/latest)
```
Comment (jlebon) [inline Jenkinsfile]: Need to escape all the `$` in this hunk.

**Comment by jlebon** [comment]
Yeah, HACKING.md should be pretty good now! It does involve some steps to get `oc cluster up` working, but that's ideally a one time thing :)

**Comment by jlebon** [comment]
(Also, thanks for this patch! :+1: )

**Comment by dustymabe** [comment]
marking as WIP

**Comment by jlebon** [comment]
This is now blocked on a workaround for https://github.com/coreos/mantle/issues/956.

**Comment by arithx** [comment]
Updated to use the [fcos_ci](https://github.com/arithx/mantle/tree/fcos_ci) branch of mantle.

Untested.

**Comment by jlebon** [comment]
I guess if one wanted to test this today, they'd use the `qemu_2_electric_boogaloo` branch right?

File: Jenkinsfile [hunk 2]
```diff
@@ -61,6 +61,28 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
...
+            cd mantle
+            git checkout fcos_ci
+            ./build kola kolet
+            """)
+            
+            // determine the image & run kola
+            utils.shwrap("""
+            latest_build=\$(readlink builds/latest)
+            qcow=\$(ls builds/"\${latest_build}"/*-"\${latest_build}"-qemu.qcow2)
+            mantle/bin/kola -p unprivileged-qemu --qemu-image "\${qemu}" -b fcos run | tee
```
Comment (cgwalters) [inline Jenkinsfile]: IMO something like this should be [coreos-assembler kola](https://github.com/coreos/coreos-assembler/pull/85).

File: Jenkinsfile [hunk 3]
```diff
@@ -61,6 +61,28 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
             currentBuild.description = "⚡ ${newBuildID}"
         }
 
+        stage('Test') {
+            // clone & build kola
+            utils.shwrap("""
+            git clone https://github.com/arithx/mantle
```
Comment (cgwalters) [inline Jenkinsfile]: Yeah...we need to make this more convenient in cosa.

**Comment by jlebon** [comment]
So, https://github.com/coreos/coreos-assembler/pull/85 was merged now. Rebasing on top of that would be a great opportunity for someone who wants to get familiar with the pipeline and hacking on it!

At least as a first step, just running `qemu-unpriv` as a sanity check, even if we don't act on it. And then as a follow-up we can do the "skip remaining artifacts and archive qcow2" bits.

**Pull request closed without merging**